R/pvals.R

```diff
@@ -0,0 +1,47 @@
+##' Calculate p-values adjusting for changes in sign as described by Nixon
+##' et al. (2024) in Beyond Normalization: Incorporating Scale Uncertainty
+##' in Microbiome and Gene Expression Analysis (internal only)
+##'
+##' @param p.lower A P x D x S matrix for P covariates, D taxa/genes, and S
+##'   monte carlo samples representing the lower tail p.values
+##' @param p.lower A P x D x S matrix for P covariates, D taxa/genes, and S
+##'   monte carlo samples representing the upper tail p.values
+##' @param p.adjust.method An adjutment method for p.adjust
+##' @return A list with P x D matrices with the non-adjusted and adjusted
+##'   p-values.
+##' @author Justin Silverman, Kyle McGovern
+aldex.pvals <- function(p.lower, p.upper, p.adjust.method) {
```

Comment:
there is a copy happening when you call p.lower and p.upper. Not sure, but a solution (if this function will always be internal), is to just refer to p.lower and p.upper in parent namespace? Saves a copy.